data/transactions/logic/assembler.go

```diff
@@ -786,60 +778,73 @@ func assembleSubstring(ops *OpStream, spec *OpSpec, args []string) error {
 	return nil
 }
 
-func assembleTxn(ops *OpStream, spec *OpSpec, args []string) error {
-	if len(args) != 1 {
-		return ops.error("txn expects one argument")
-	}
-	fs, ok := txnFieldSpecByName[args[0]]
+func txnFieldImm(name string, expectArray bool, ops *OpStream) (*txnFieldSpec, error) {
+	fs, ok := TxnFieldSpecByName[name]
 	if !ok {
-		return ops.errorf("txn unknown field: %#v", args[0])
+		return nil, fmt.Errorf("unknown field: %#v", name)
 	}
-	_, ok = txnaFieldSpecByField[fs.field]
-	if ok {
-		return ops.errorf("found array field %#v in txn op", args[0])
+	if expectArray != fs.array {
+		if expectArray {
+			return nil, fmt.Errorf("found scalar field %#v while expecting array", name)
+		}
+		return nil, fmt.Errorf("found array field %#v while expecting scalar", name)
 	}
 	if fs.version > ops.Version {
-		return ops.errorf("field %#v available in version %d. Missed #pragma version?", args[0], fs.version)
+		return nil,
+			fmt.Errorf("field %#v available in version %d. Missed #pragma version?", name, fs.version)
```

Comment:
I think that this message could be improved - we can definitely tell if the user had a pragma or not, and if so, what version he selected. That being said, it shouldn't be a blocker, and we can improve this later on.

Reply:
I agree, but I'm going to defer this unless you disagree.